src/time_integration/paired_explicit_runge_kutta/polynomial_optimizer.jl

```diff
@@ -0,0 +1,138 @@
+using LinearAlgebra: eigvals
+using ECOS
+using Convex
+const MOI = Convex.MOI
+
+function filter_eigvals(eig_vals, threshold)
+    filtered_eigvals_counter = 0
+    filtered_eig_vals = Complex{Float64}[]
+
+    for eig_val in eig_vals
+        # Filter out eigenvalues with positive real parts, those with negative imaginary parts due to eigenvalues' symmetry
+        # around real axis, or the eigenvalues that are too small.
+        if real(eig_val) > 0 || imag(eig_val) < 0 || abs(eig_val) < threshold
+            filtered_eigvals_counter += 1
+        else
+            push!(filtered_eig_vals, eig_val)
+        end
+    end
+
+    println("$filtered_eigvals_counter eigenvalue(s) are not passed on because " *
+            "they either are in magnitude smaller than $threshold, have positive " *
+            "real parts, or have negative imaginary parts.\n")
+
+    return length(filtered_eig_vals), filtered_eig_vals
+end
+
+function stability_polynomials(cons_order, num_stage_evals, num_eig_vals,
+                               normalized_powered_eigvals_scaled, pnoms, gamma::Variable)
+    # Initialize with zero'th order (z^0) coefficient
+    for i in 1:num_eig_vals
+        pnoms[i] = 1.0
+    end
+
+    # First `cons_order` terms of the exponential
+    for k in 1:cons_order
+        for i in 1:num_eig_vals
+            pnoms[i] += normalized_powered_eigvals_scaled[i, k]
+        end
+    end
+
+    # Contribution from free coefficients
+    for k in (cons_order + 1):num_stage_evals
+        pnoms += gamma[k - cons_order] * normalized_powered_eigvals_scaled[:, k]
+    end
+
+    # For optimization only the maximum is relevant
+    return maximum(abs(pnoms))
+end
+
+"""
+    bisection()
+
+The following structures and methods provide a simplified implementation to 
+discover optimal stability polynomial for a given set of `eig_vals`
+These are designed for the one-step (i.e., Runge-Kutta methods) integration of initial value ordinary 
+and partial differential equations.
+
+- Ketcheson and Ahmadia (2012).
+Optimal stability polynomials for numerical integration of initial value problems
+[DOI: 10.2140/camcos.2012.7.247](https://doi.org/10.2140/camcos.2012.7.247)
+"""
+
+function bisection(cons_order, num_eig_vals, num_stage_evals, dtmax, dteps, eig_vals)
```

Comment:
Docstring detached from function

```suggestion
"""
function bisection(cons_order, num_eig_vals, num_stage_evals, dtmax, dteps, eig_vals)
```